src/acl/DestinationDomain.cc

```diff
@@ -16,28 +16,21 @@
 #include "fqdncache.h"
 #include "HttpRequest.h"
 
-DestinationDomainLookup DestinationDomainLookup::instance_;
+static void LookupDone(const char *, const Dns::LookupDetails &, void *data);
 
-DestinationDomainLookup *
-DestinationDomainLookup::Instance()
+static void
+StartLookup(ACLFilledChecklist &cl, const ACL &)
 {
-    return &instance_;
+    fqdncache_nbgethostbyaddr(cl.dst_addr, LookupDone, &cl);
```

Comment:
The diff does not make it obvious, but we were declaring a hierarchy of AsyncState classes just to make async function calls like this fqdncache_nbgethostbyaddr() call. Those classes are now replaced by simple static functions like this StartLookup(), simplifying Checklist logic quite a bit!

In the foreseeable future, we would be able to get rid of those static functions as well, but that requires improving all async functions like fqdncache_nbgethostbyaddr() to become fully async, which requires non-trivial work that deserves dedicated PRs. Today, those functions sometimes call their callbacks immediately, before returning to the caller, creating all sorts of problems.